Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts

```diff
@@ -0,0 +1,156 @@
+        if (result && result['artifacts']) {
+            let artifacts = result['artifacts'];
+            if (artifacts.length === 0) {
+                tl.setResult(tl.TaskResult.Failed, tl.loc('CouldNotFindArtifacts', jobName, jobBuildId));
+            } else {
+                let articactsRelativePath = result['artifacts'].map(artifact => {
+                    return artifact['relativePath'];
+                });
+                defer.resolve(articactsRelativePath);
+            }
+        } else {
+            // no artifacts for this job
+            tl.setResult(tl.TaskResult.Failed, tl.loc('CouldNotFindArtifacts', jobName, jobBuildId));
+        }
+    });
+
+    return defer.promise 
+}
+
+async function download(url: string, localFile: string, cred: Credential, strictSSL: boolean) {
+    tl.debug(tl.loc("DownloadFileTo", url, localFile));
+    await request.get( {url: url, strictSSL: strictSSL})
+        .auth(cred.mUsername, cred.mPassword, true)
+        .on('error', err => {
+            tl.setResult(tl.TaskResult.Failed, err)
+        })
+        .pipe(fs.createWriteStream(localFile));
+
+    tl.debug(tl.loc("FileSuccessfullyDownloaded", localFile));
+}
+
+function getArtifactUrl(serverEndpointUrl: string, jobName: string, jobBuildId: number, relativePath: string) {
+    return `${serverEndpointUrl}/job/${jobName}/${jobBuildId}/artifact/${relativePath}`;
+}
+
+async function doWork() {
+    try {
+        tl.setResourcePath(path.join( __dirname, 'task.json'));
+
+        let serverEndpoint: string = tl.getInput('serverEndpoint', true);
+        let serverEndpointUrl: string = tl.getEndpointUrl(serverEndpoint, false);
+
+        let serverEndpointAuth: tl.EndpointAuthorization = tl.getEndpointAuthorization(serverEndpoint, false);
+        let username: string = serverEndpointAuth['parameters']['username'];
+        let password: string = serverEndpointAuth['parameters']['password'];
+        let cred: Credential = (username && password) ? new Credential(username, password) : new Credential("", "");
+        
+        let jobName: string = tl.getInput('jobName', true);
+        let localPathRoot: string = tl.getPathInput('saveTo', true);
+
+        let strictSSL: boolean = !(tl.getBoolInput("trustSSL", false));
+
+        let jenkinsBuild: string = tl.getInput("jenkinsBuild", true);
+        let buildId: number;
+        if (jenkinsBuild === "LastSuccessfulBuild") {
+            tl.debug(tl.loc('GetArtifactsFromLastSuccessfulBuild', jobName));
+            buildId = await getLastSuccessful(serverEndpointUrl, jobName, cred, strictSSL);
+        } else {
+            let buildIdStr = tl.getInput("jenkinsBuildNumber");
+            buildId = parseInt(buildIdStr);
+        }
+        tl.debug(tl.loc('GetArtifactsFromBuildNumber', buildId, jobName));
+        
+        let artifactsRelativePaths = await getArtifactsRelativePaths(serverEndpointUrl, jobName, 
+                buildId, cred, strictSSL);
```

Comment:
This is a continuation of the prior line which is a method call, so I opted to indent with 8 spaces to indicate that.